packages/reason-relay/reason-relay-ppx/library/Query.re

```diff
@@ -0,0 +1,356 @@
+              ->[%e valFromGeneratedModule(["Internal", "convertVariables"])]
+              ->ReasonRelay_Internal.internal_cleanObjectFromUndefinedRaw,
+              {
+                fetchKey,
+                fetchPolicy: fetchPolicy->ReasonRelay.mapFetchPolicy,
+                renderPolicy: renderPolicy->ReasonRelay.mapRenderPolicy,
+                networkCacheConfig,
+              },
+            );
+
+          ReasonRelay_Internal.internal_useConvertedValue(
+            [%e valFromGeneratedModule(["Internal", "convertResponse"])],
+            data,
+          );
+        }
+      ],
+      [%stri
+        let useLoader =
+            ()
+            : (
+                option([%t typeFromGeneratedModule(["queryRef"])]),
+                (
+                  ~variables: [%t
+                                typeFromGeneratedModule([
+                                  "Types",
+                                  "variables",
+                                ])
+                              ],
+                  ~fetchPolicy: ReasonRelay.fetchPolicy=?,
+                  ~networkCacheConfig: ReasonRelay.cacheConfig=?,
+                  unit
+                ) =>
+                unit,
+                unit => unit,
+              ) => {
+          let (nullableQueryRef, loadQueryFn, disposableFn) =
+            Internal.internal_useQueryLoader(
+              [%e valFromGeneratedModule(["node"])],
+            );
+
+          // TODO: Fix stability of this reference. Can't seem to use React.useCallback with labelled arguments for some reason.
+          let loadQuery =
+              (
+                ~variables: [%t
+                   typeFromGeneratedModule(["Types", "variables"])
+                 ],
+                ~fetchPolicy=?,
+                ~networkCacheConfig=?,
+                (),
+              ) =>
+            loadQueryFn(
+              variables->[%e
+                           valFromGeneratedModule([
+                             "Internal",
+                             "convertVariables",
+                           ])
+                         ],
+              {fetchPolicy, networkCacheConfig},
+            );
+
+          (Js.Nullable.toOption(nullableQueryRef), loadQuery, disposableFn);
+        }
+      ],
+      [%stri
+        let fetch =
+            (
+              ~environment: ReasonRelay.Environment.t,
+              ~variables: [%t
+                 typeFromGeneratedModule(["Types", "variables"])
+               ],
+              ~onResult:
+                 Belt.Result.t(
+                   [%t typeFromGeneratedModule(["Types", "response"])],
+                   Js.Exn.t,
+                 ) =>
+                 unit,
+              ~networkCacheConfig: option(ReasonRelay.cacheConfig)=?,
+              ~fetchPolicy: option(ReasonRelay.fetchQueryFetchPolicy)=?,
+              (),
+            )
+            : unit => {
+          let _ =
+            Internal.internal_fetchQuery(
```

Comment:
Will this not cause a memory leak? I am concerned that this does not return the subscription to because to cancel.

Reply:
I don't think it'll cause a memory leak as the observable for `fetchQuery` will just complete itself after a response. However,  I might as well return the disposable here, there's no reason not to. Good point! 👍